ember-scoped-css/src/lib/scoped-css-preprocessor.js

```diff
@@ -67,12 +71,57 @@ class ScopedFilter extends Filter {
       return '';
     }
   }
+  async postProcess(results, relativePath) {
+    if (relativePath.endsWith('.module.css')) {
+      return results;
+    }
+
+    // need to add gjs/gts support
+    // check existence of gjs and gts files
+    // extract template tag
+
+    const templatePath = relativePath.replace(/\.css/, '.hbs');
+
+    for (let inputPath of this.inputPaths) {
+      const templateFilePath = path.join(inputPath, templatePath);
+      const fileExists = await fsExists(templateFilePath);
+
+      if (fileExists) {
+        const cssFilePath = path.join(inputPath, relativePath);
+        const cssContents = await readFile(cssFilePath,  'utf-8');
+        const { classes, tags } = getClassesTagsFromCss(cssContents);
+
+        const previousClasses = this.options.previousClasses.get(relativePath);
+
+        // if we have previous classes, and they are different, build templates to compare
+        if (previousClasses && classes.toString() !== previousClasses.toString()) {
+          const localPackagerStylePath = packageScopedPathToModulePath(relativePath);
+          const postfix = generateHash(localPackagerStylePath);
+          const templateContents = await readFile(templateFilePath,  'utf-8');
+          const templateOriginal = rewriteHbs(templateContents, previousClasses, tags, postfix);
+          const templateRewrite = rewriteHbs(templateContents, classes, tags, postfix);
+          // if the rewrite doesn't match the original output, we want to rebuild the template
+          if (templateOriginal !== templateRewrite) {
+            // this is an awful hack because we don't know how to invalidate broccoli-persistent-filter cache
+            // ideally we'd invalidate the broccoli-peristent-filter cache here
+            await writeFile(templateFilePath, templateContents + ' ');
```

Comment:
I feel sad that I had to revert to using a hack like this, but the cache invalidation was not working as advertised (or I was just doing it wrong). Would love to fix this to actually invalidate the cache correctly, but this will work for now.